### PR TITLE
Only add PointerGesture if PointerOver VisualState exists

### LINKED ISSUE #11157: [Android] ScrollView accessibility is broken
### Description

There's currently an accessibility bug where a ScrollView is in the accessibility tree and receives screen reader focus, and the non-interactive elements within a ScrollView are not individually accessible. 

This is a regression - it's broken with .NET 7 and works fine with .NET 6.
It affects our File > New .NET MAUI app template experience.

| Current behavior | Expected behavior |
|---|---|
|image|image|

This issue is very similar but different from https://github.com/dotnet/maui/issues/8524
This is an issue with .NET MAUI and note the Android platform

Current workaround is to remove the ScrollView or set `AutomationProperties.IsInAccessibleTree="False"` on the ScrollView

### Steps to Reproduce

1. Create a File > New .NET MAUI app with .NET 7
2. Run on Android
3. Turn Talkback on

### Link to public reproduction project repository

file new maui

### Version with bug

6.0.486 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 12

### Did you find any workaround?

Current workaround is to remove the ScrollView or set `AutomationProperties.IsInAccessibleTree="False"` on the ScrollView

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by PureWeen**
### Description of Change
On `Android` if you add a `HoverListener` to a `View` then `Android` will automatically consider the element an accessibility element. We were naively hopefully that just adding a `PointerGesture` to everything would be fine but alas it was not. 

image

https://github.com/dotnet/maui/issues/11157#issuecomment-1322774433

### Issues Fixed
Fixes #11157

### CODE REVIEW

**Comment by hartez** [comment]
@PureWeen  Does this actually fix the problem, or just reduce the number of instances of the problem? Won't layouts with hover states still show up as accessibility elements?

**Comment by PureWeen** [comment]
> @PureWeen Does this actually fix the problem, or just reduce the number of instances of the problem? Won't layouts with hover states still show up as accessibility elements?

I logged an issue here with my thoughts https://github.com/dotnet/maui/issues/11742.

I'm not fully satisfied with this solution. I think satisfaction needs a bit more research.
I'm hesitant to try and be too smart here, so for now I went with pulling us back from "everything is broken by default". 

I think only adding the PGR if the "PointerOver" VSM is present is valid logic to have, so I think this PR at least pushes the story into the right direction.

File: src/Controls/tests/DeviceTests/Controls.DeviceTests.csproj
Comment (rmarinho) [inline src/Controls/tests/DeviceTests/Controls.DeviceTests.csproj:20]: i don't think we need this change

Comment (PureWeen) [reply src/Controls/tests/DeviceTests/Controls.DeviceTests.csproj]: Fixed

File: src/Controls/src/Core/VisualStateManager.cs
Comment (hartez) [inline src/Controls/src/Core/VisualStateManager.cs:668]: VisualState names have to be unique, so you can just call this `FindVisualStateByName`. Or better yet, just `bool HasVisualState()`.

File: src/Controls/src/Core/PointerGestureRecognizer.cs
Comment (hartez) [inline src/Controls/src/Core/PointerGestureRecognizer.cs:185]: What happens if there's no VisualState, but the user is doing something else in response to the pointer entering the control?

File: src/Controls/src/Core/View.cs
Comment (hartez) [inline src/Controls/src/Core/View.cs:171]: Unfortunately, need to update all the public API stuff.

File: src/Controls/src/Core/VisualStateManager.cs
Comment (hartez) [inline src/Controls/src/Core/VisualStateManager.cs:670]: This should be `for` loops for performance, probably.

**Review by hartez — CHANGES_REQUESTED** [review]

File: src/Controls/src/Core/PointerGestureRecognizer.cs
Comment (PureWeen) [reply src/Controls/src/Core/PointerGestureRecognizer.cs:185]: This is a pointer gesture that's hidden from the user via `IGestureController`. `CompositeGestureRecognizers` isn't something that's exposed to the user, so if the user adds a `PointerGestureRecognizer` it will still remain. This is only adding and removing the `PGR` that we are adding in response to the existence of a `PointerOver` VSM state.

File: src/Controls/src/Core/VisualStateManager.cs
Comment (rachelkang) [inline src/Controls/src/Core/VisualStateManager.cs:680]:
```suggestion
		return state.Name == name;
```

File: src/Controls/tests/Core.UnitTests/PointerGestureRecognizerTests.cs
Comment (rachelkang) [inline src/Controls/tests/Core.UnitTests/PointerGestureRecognizerTests.cs:18]: can't actually remember the acronym rule myself, but can we double check to see which is relevant here?

```suggestion
		public void PointerOverVsmGestureIsOnlyPresentWithVsm()
```

File: src/Controls/tests/DeviceTests/ControlsDeviceTestExtensions.cs
Comment (rachelkang) [inline src/Controls/tests/DeviceTests/ControlsDeviceTestExtensions.cs:27]: I think you opened an issue for this. Could you link to the issue in the comment here?

File: src/Controls/tests/DeviceTests/ControlsDeviceTestExtensions.cs
Comment (rachelkang) [inline src/Controls/tests/DeviceTests/ControlsDeviceTestExtensions.cs:32]: I'm a bit lost on this if statement. Could you explain the `!expected.Value` part to me?

File: src/Controls/tests/DeviceTests/Elements/Accessibility/AccessibilityTests.cs
Comment (rachelkang) [inline src/Controls/tests/DeviceTests/Elements/Accessibility/AccessibilityTests.cs:134]: Are all of these if/defs based on your observations of the current iOS default platform-specifci behaviors?

File: src/Controls/tests/DeviceTests/Elements/Accessibility/AccessibilityTests.cs
Comment (rachelkang) [inline src/Controls/tests/DeviceTests/Elements/Accessibility/AccessibilityTests.cs:83]: Why RefreshView and why only RefreshView here? (as opposed to also other layouts)

File: src/TestUtils/src/DeviceTests/AssertionExtensions.iOS.cs
Comment (rachelkang) [inline src/TestUtils/src/DeviceTests/AssertionExtensions.iOS.cs:466]:
```suggestion
				// unless you have VoiceOver turned on.
```

File: src/TestUtils/src/DeviceTests/AssertionExtensions.iOS.cs
Comment (rachelkang) [inline src/TestUtils/src/DeviceTests/AssertionExtensions.iOS.cs:469]:
```suggestion
				// According to the Apple docs anything that inherits from UIControl
```

File: src/TestUtils/src/DeviceTests/AssertionExtensions.iOS.cs
Comment (rachelkang) [inline src/TestUtils/src/DeviceTests/AssertionExtensions.iOS.cs:478]:
```suggestion
				// I tested each of these controls inside Xcode away from any MAUI tampering
```

**Review by rachelkang — COMMENTED** [review]
Amazing!!! Thanks for deep diving into this and getting this first fix through the door. 
ScrollView no longer receives screen reader focus by default on Android, and Pointer gesture and PointerOver state both work as expected on all platforms.

I left a couple nitpicky comments and questions for my own curiosity, but overall this should be good to go soon!

File: src/Controls/tests/DeviceTests/Elements/Accessibility/AccessibilityTests.cs
Comment (PureWeen) [reply src/Controls/tests/DeviceTests/Elements/Accessibility/AccessibilityTests.cs:134]: Yes, I also tested each of these in a basic Xcode project to verify the same behavior.

File: src/Controls/tests/DeviceTests/ControlsDeviceTestExtensions.cs
Comment (PureWeen) [reply src/Controls/tests/DeviceTests/ControlsDeviceTestExtensions.cs:27]: Are you thinking of this issue?
https://github.com/dotnet/maui/issues/11762

It's not really related. This comment just explains why I have all this odd-looking code here and why none of this is really that accurate unless you run these on a physical device

File: src/Controls/tests/DeviceTests/Elements/Accessibility/AccessibilityTests.cs
Comment (PureWeen) [reply src/Controls/tests/DeviceTests/Elements/Accessibility/AccessibilityTests.cs]: Aw yea, let me remove this. I was working around a bug with `RefreshView` that was recently fixed https://github.com/dotnet/maui/pull/11735

File: src/Controls/src/Core/VisualStateManager.cs
Comment (hartez) [inline src/Controls/src/Core/VisualStateManager.cs]: This will return `false` on the first non-matching state name.

**Review by hartez — CHANGES_REQUESTED** [review]
Not sure about the implementation of HasVisualState

File: src/Controls/src/Core/VisualStateManager.cs
Comment (PureWeen) [reply src/Controls/src/Core/VisualStateManager.cs]: Derp :-/ fixed

File: src/Controls/tests/DeviceTests/ControlsDeviceTestExtensions.cs
Comment (PureWeen) [reply src/Controls/tests/DeviceTests/ControlsDeviceTestExtensions.cs]: Deleted this extension method and reworked the non `VoiceOver` path

File: src/Controls/src/Core/VisualStateManager.cs
Comment (rachelkang) [reply src/Controls/src/Core/VisualStateManager.cs]: sorry for the confusion!

**Review by rachelkang — APPROVED** [review]
LGTM! Once it's all green, I think it's good to merge :)

**Review by hartez — APPROVED** [review]

**Review by Eilon — APPROVED** [review]

**Pull request merged**
